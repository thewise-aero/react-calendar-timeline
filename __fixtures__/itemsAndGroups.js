
export const items = [
  {
    id: '0',
    group: '1',
    start_time: 1540540000000,
    end_time: 1540546803877,
    canMove: false,
    canResize: false,
    title: 'title 0',
  },
  {
    id: '5',
    group: '1',
    start_time: 1540532800000,
    end_time: 1540554003877,
    canMove: false,
    canResize: false,
    className: '',
    title: 'title 5',
  },
  {
    id: '6',
    group: '1',
    start_time: 1540550800000,
    end_time: 1540575603877,
    canMove: false,
    canResize: false,
    className: '',
    title: 'title 6',
  },
  {
    id: '1',
    group: '1',
    start_time: 1540570000000,
    end_time: 1540584875919,
    canMove: true,
    canResize: 'both',
    title: 'title 1',
  },
  {
    id: '2',
    group: '1',
    start_time: 1540620000000,
    end_time: 1540640397548,
    canMove: false,
    canResize: false,
    className: '',
    title: 'title 2',
  },
  {
    id: '3',
    group: '3',
    start_time: 1540656000000,
    end_time: 1540676397548,
    canMove: false,
    canResize: false,
    className: '',
    title: 'title 3',
  }
]

export const groups = [{ id: '1' }, { id: '2' }, { id: '3' }]
