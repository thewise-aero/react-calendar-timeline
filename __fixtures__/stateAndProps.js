import { defaultKeys } from 'lib/default-config'
import {items} from './itemsAndGroups'
import Timeline from 'lib/Timeline'

export const visibleTimeStart = 1540501200000
export const visibleTimeEnd = 1540587600000

export const props = {
  ...Timeline.defaultProps,
  keys: defaultKeys,
  lineHeight: 30,
  stackItems: true,
  itemHeightRatio: 0.75,
  visibleTimeEnd,
  visibleTimeStart,
}

export const propsNoStack = {
  ...props,
  stackItems: false,
}


export const state = {
  draggingItem: undefined,
  dragTime: null,
  resizingItem: null,
  resizingEdge: null,
  resizeTime: null,
  newGroupId: null,
  selectedItem: null,
  canvasTimeStart: 1540414800000,
  visibleTimeEnd: visibleTimeEnd,
  visibleTimeStart: visibleTimeStart,
  canvasTimeEnd: 1540674000000,
  width: 1000,
}

//offset 1 hour
const timeOffset = 1 * 60 *60 *1000

export const stateMoveItem = {
  ...state,
  draggingItem: items[0].id,
  dragTime: items[0].start_time+timeOffset,
  newGroupId: items[0].group,
}
export const stateResizeItemLeft = {
  ...state,
  resizingItem: items[0].id,
  resizingEdge: 'left',
  resizeTime: items[0].start_time+timeOffset,
  newGroupId: items[0].group,
}

export const stateResizeItemRight = {
  ...state,
  resizingItem: items[0].id,
  resizingEdge: 'right',
  resizeTime: items[0].end_time+timeOffset,
  newGroupId: items[0].group,
}