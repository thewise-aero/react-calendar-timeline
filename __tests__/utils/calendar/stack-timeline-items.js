import { stackTimelineItems } from 'lib/utility/calendar'
import { items, groups } from '../../../__fixtures__/itemsAndGroups'
import {
  props,
  state,
  stateMoveItem,
  stateResizeItemLeft,
  stateResizeItemRight,
  propsNoStack,
} from '../../../__fixtures__/stateAndProps'
describe('stackItems', () => {
  it('work as expected', () => {
    expect(
      stackTimelineItems(
        items,
        groups,
        9000,
        state.canvasTimeStart,
        state.canvasTimeEnd,
        props.keys,
        props.lineHeight,
        props.itemHeightRatio,
        props.stackItems,
        state.draggingItem,
        state.resizingItem,
        state.dragTime,
        state.resizingEdge,
        state.resizeTime,
        state.newGroupId
      )
    ).toMatchSnapshot()
  })
  it('work as expected no stack', () => {
    expect(
      stackTimelineItems(
        items,
        groups,
        9000,
        state.canvasTimeStart,
        state.canvasTimeEnd,
        propsNoStack.keys,
        propsNoStack.lineHeight,
        propsNoStack.itemHeightRatio,
        propsNoStack.stackItems,
        state.draggingItem,
        state.resizingItem,
        state.dragTime,
        state.resizingEdge,
        state.resizeTime,
        state.newGroupId
      )
    ).toMatchSnapshot()
  })
  it('should stack items while moving an item', () => {
    expect(
      stackTimelineItems(
        items,
        groups,
        9000,
        stateMoveItem.canvasTimeStart,
        stateMoveItem.canvasTimeEnd,
        props.keys,
        props.lineHeight,
        props.itemHeightRatio,
        props.stackItems,
        stateMoveItem.draggingItem,
        stateMoveItem.resizingItem,
        stateMoveItem.dragTime,
        stateMoveItem.resizingEdge,
        stateMoveItem.resizeTime,
        stateMoveItem.newGroupId
      )
    ).toMatchSnapshot()
  })
  it('should stack items while resize item left', () => {
    expect(
      stackTimelineItems(
        items,
        groups,
        9000,
        stateResizeItemLeft.canvasTimeStart,
        stateResizeItemLeft.canvasTimeEnd,
        props.keys,
        props.lineHeight,
        props.itemHeightRatio,
        props.stackItems,
        stateResizeItemLeft.draggingItem,
        stateResizeItemLeft.resizingItem,
        stateResizeItemLeft.dragTime,
        stateResizeItemLeft.resizingEdge,
        stateResizeItemLeft.resizeTime,
        stateResizeItemLeft.newGroupId
      )
    ).toMatchSnapshot()
  })
  it('should stack items while resize item right', () => {
    expect(
      stackTimelineItems(
        items,
        groups,
        9000,
        stateResizeItemRight.canvasTimeStart,
        stateResizeItemRight.canvasTimeEnd,
        props.keys,
        props.lineHeight,
        props.itemHeightRatio,
        props.stackItems,
        stateResizeItemRight.draggingItem,
        stateResizeItemRight.resizingItem,
        stateResizeItemRight.dragTime,
        stateResizeItemRight.resizingEdge,
        stateResizeItemRight.resizeTime,
        stateResizeItemRight.newGroupId
      )
    ).toMatchSnapshot()
  })
  it('should return empty dimensions if groups are empty', () => {
    expect(
      stackTimelineItems(
        items,
        [],
        9000,
        state.canvasTimeStart,
        state.canvasTimeEnd,
        props.keys,
        props.lineHeight,
        props.itemHeightRatio,
        props.stackItems,
        state.draggingItem,
        state.resizingItem,
        state.dragTime,
        state.resizingEdge,
        state.resizeTime,
        state.newGroupId
      )
    ).toMatchObject({
      groupsWithItemsDimensions: {},
      height: 0,
      groupHeights: [],
      groupTops: []
    })
  })
})
