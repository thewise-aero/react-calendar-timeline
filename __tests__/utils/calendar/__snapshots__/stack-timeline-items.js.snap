// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stackItems should stack items while moving an item 1`] = `
Object {
  "groupHeights": Array [
    60,
    30,
    30,
  ],
  "groupTops": Array [
    0,
    60,
    90,
  ],
  "groupsWithItemsDimensions": Object {
    "1": Object {
      "group": Object {
        "id": "1",
      },
      "height": 60,
      "index": 0,
      "itemDimensions": Array [
        Object {
          "dimensions": Object {
            "collisionLeft": 1540543600000,
            "collisionWidth": 6803877,
            "height": 22.5,
            "left": 4472.222222222223,
            "stack": true,
            "top": 3.75,
            "width": 236.24572916666602,
          },
          "id": "0",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540532800000,
            "collisionWidth": 21203877,
            "height": 22.5,
            "left": 4097.222222222223,
            "stack": true,
            "top": 33.75,
            "width": 736.245729166666,
          },
          "id": "5",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540550800000,
            "collisionWidth": 24803877,
            "height": 22.5,
            "left": 4722.222222222223,
            "stack": true,
            "top": 3.75,
            "width": 861.245729166666,
          },
          "id": "6",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540570000000,
            "collisionWidth": 14875919,
            "height": 22.5,
            "left": 5388.888888888889,
            "stack": true,
            "top": 33.75,
            "width": 516.5249652777784,
          },
          "id": "1",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540620000000,
            "collisionWidth": 20397548,
            "height": 22.5,
            "left": 7125,
            "stack": true,
            "top": 3.75,
            "width": 708.2481944444444,
          },
          "id": "2",
        },
      ],
      "items": Array [
        Object {
          "canMove": false,
          "canResize": false,
          "end_time": 1540550403877,
          "group": "1",
          "id": "0",
          "start_time": 1540543600000,
          "title": "title 0",
        },
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540554003877,
          "group": "1",
          "id": "5",
          "start_time": 1540532800000,
          "title": "title 5",
        },
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540575603877,
          "group": "1",
          "id": "6",
          "start_time": 1540550800000,
          "title": "title 6",
        },
        Object {
          "canMove": true,
          "canResize": "both",
          "end_time": 1540584875919,
          "group": "1",
          "id": "1",
          "start_time": 1540570000000,
          "title": "title 1",
        },
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540640397548,
          "group": "1",
          "id": "2",
          "start_time": 1540620000000,
          "title": "title 2",
        },
      ],
    },
    "2": Object {
      "group": Object {
        "id": "2",
      },
      "height": 30,
      "index": 1,
      "itemDimensions": Array [],
      "items": Array [],
    },
    "3": Object {
      "group": Object {
        "id": "3",
      },
      "height": 30,
      "index": 2,
      "itemDimensions": Array [
        Object {
          "dimensions": Object {
            "collisionLeft": 1540656000000,
            "collisionWidth": 20397548,
            "height": 22.5,
            "left": 8375,
            "stack": true,
            "top": 3.75,
            "width": 625,
          },
          "id": "3",
        },
      ],
      "items": Array [
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540676397548,
          "group": "3",
          "id": "3",
          "start_time": 1540656000000,
          "title": "title 3",
        },
      ],
    },
  },
  "height": 120,
  "itemsWithInteractions": Array [
    Object {
      "canMove": false,
      "canResize": false,
      "end_time": 1540550403877,
      "group": "1",
      "id": "0",
      "start_time": 1540543600000,
      "title": "title 0",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540554003877,
      "group": "1",
      "id": "5",
      "start_time": 1540532800000,
      "title": "title 5",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540575603877,
      "group": "1",
      "id": "6",
      "start_time": 1540550800000,
      "title": "title 6",
    },
    Object {
      "canMove": true,
      "canResize": "both",
      "end_time": 1540584875919,
      "group": "1",
      "id": "1",
      "start_time": 1540570000000,
      "title": "title 1",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540640397548,
      "group": "1",
      "id": "2",
      "start_time": 1540620000000,
      "title": "title 2",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540676397548,
      "group": "3",
      "id": "3",
      "start_time": 1540656000000,
      "title": "title 3",
    },
  ],
}
`;

exports[`stackItems should stack items while resize item left 1`] = `
Object {
  "groupHeights": Array [
    60,
    30,
    30,
  ],
  "groupTops": Array [
    0,
    60,
    90,
  ],
  "groupsWithItemsDimensions": Object {
    "1": Object {
      "group": Object {
        "id": "1",
      },
      "height": 60,
      "index": 0,
      "itemDimensions": Array [
        Object {
          "dimensions": Object {
            "collisionLeft": 1540543600000,
            "collisionWidth": 3203877,
            "height": 22.5,
            "left": 4472.222222222223,
            "stack": true,
            "top": 3.75,
            "width": 111.24572916666602,
          },
          "id": "0",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540532800000,
            "collisionWidth": 21203877,
            "height": 22.5,
            "left": 4097.222222222223,
            "stack": true,
            "top": 33.75,
            "width": 736.245729166666,
          },
          "id": "5",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540550800000,
            "collisionWidth": 24803877,
            "height": 22.5,
            "left": 4722.222222222223,
            "stack": true,
            "top": 3.75,
            "width": 861.245729166666,
          },
          "id": "6",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540570000000,
            "collisionWidth": 14875919,
            "height": 22.5,
            "left": 5388.888888888889,
            "stack": true,
            "top": 33.75,
            "width": 516.5249652777784,
          },
          "id": "1",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540620000000,
            "collisionWidth": 20397548,
            "height": 22.5,
            "left": 7125,
            "stack": true,
            "top": 3.75,
            "width": 708.2481944444444,
          },
          "id": "2",
        },
      ],
      "items": Array [
        Object {
          "canMove": false,
          "canResize": false,
          "end_time": 1540546803877,
          "group": "1",
          "id": "0",
          "start_time": 1540543600000,
          "title": "title 0",
        },
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540554003877,
          "group": "1",
          "id": "5",
          "start_time": 1540532800000,
          "title": "title 5",
        },
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540575603877,
          "group": "1",
          "id": "6",
          "start_time": 1540550800000,
          "title": "title 6",
        },
        Object {
          "canMove": true,
          "canResize": "both",
          "end_time": 1540584875919,
          "group": "1",
          "id": "1",
          "start_time": 1540570000000,
          "title": "title 1",
        },
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540640397548,
          "group": "1",
          "id": "2",
          "start_time": 1540620000000,
          "title": "title 2",
        },
      ],
    },
    "2": Object {
      "group": Object {
        "id": "2",
      },
      "height": 30,
      "index": 1,
      "itemDimensions": Array [],
      "items": Array [],
    },
    "3": Object {
      "group": Object {
        "id": "3",
      },
      "height": 30,
      "index": 2,
      "itemDimensions": Array [
        Object {
          "dimensions": Object {
            "collisionLeft": 1540656000000,
            "collisionWidth": 20397548,
            "height": 22.5,
            "left": 8375,
            "stack": true,
            "top": 3.75,
            "width": 625,
          },
          "id": "3",
        },
      ],
      "items": Array [
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540676397548,
          "group": "3",
          "id": "3",
          "start_time": 1540656000000,
          "title": "title 3",
        },
      ],
    },
  },
  "height": 120,
  "itemsWithInteractions": Array [
    Object {
      "canMove": false,
      "canResize": false,
      "end_time": 1540546803877,
      "group": "1",
      "id": "0",
      "start_time": 1540543600000,
      "title": "title 0",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540554003877,
      "group": "1",
      "id": "5",
      "start_time": 1540532800000,
      "title": "title 5",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540575603877,
      "group": "1",
      "id": "6",
      "start_time": 1540550800000,
      "title": "title 6",
    },
    Object {
      "canMove": true,
      "canResize": "both",
      "end_time": 1540584875919,
      "group": "1",
      "id": "1",
      "start_time": 1540570000000,
      "title": "title 1",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540640397548,
      "group": "1",
      "id": "2",
      "start_time": 1540620000000,
      "title": "title 2",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540676397548,
      "group": "3",
      "id": "3",
      "start_time": 1540656000000,
      "title": "title 3",
    },
  ],
}
`;

exports[`stackItems should stack items while resize item right 1`] = `
Object {
  "groupHeights": Array [
    60,
    30,
    30,
  ],
  "groupTops": Array [
    0,
    60,
    90,
  ],
  "groupsWithItemsDimensions": Object {
    "1": Object {
      "group": Object {
        "id": "1",
      },
      "height": 60,
      "index": 0,
      "itemDimensions": Array [
        Object {
          "dimensions": Object {
            "collisionLeft": 1540540000000,
            "collisionWidth": 10403877,
            "height": 22.5,
            "left": 4347.222222222223,
            "stack": true,
            "top": 3.75,
            "width": 361.245729166666,
          },
          "id": "0",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540532800000,
            "collisionWidth": 21203877,
            "height": 22.5,
            "left": 4097.222222222223,
            "stack": true,
            "top": 33.75,
            "width": 736.245729166666,
          },
          "id": "5",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540550800000,
            "collisionWidth": 24803877,
            "height": 22.5,
            "left": 4722.222222222223,
            "stack": true,
            "top": 3.75,
            "width": 861.245729166666,
          },
          "id": "6",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540570000000,
            "collisionWidth": 14875919,
            "height": 22.5,
            "left": 5388.888888888889,
            "stack": true,
            "top": 33.75,
            "width": 516.5249652777784,
          },
          "id": "1",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540620000000,
            "collisionWidth": 20397548,
            "height": 22.5,
            "left": 7125,
            "stack": true,
            "top": 3.75,
            "width": 708.2481944444444,
          },
          "id": "2",
        },
      ],
      "items": Array [
        Object {
          "canMove": false,
          "canResize": false,
          "end_time": 1540550403877,
          "group": "1",
          "id": "0",
          "start_time": 1540540000000,
          "title": "title 0",
        },
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540554003877,
          "group": "1",
          "id": "5",
          "start_time": 1540532800000,
          "title": "title 5",
        },
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540575603877,
          "group": "1",
          "id": "6",
          "start_time": 1540550800000,
          "title": "title 6",
        },
        Object {
          "canMove": true,
          "canResize": "both",
          "end_time": 1540584875919,
          "group": "1",
          "id": "1",
          "start_time": 1540570000000,
          "title": "title 1",
        },
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540640397548,
          "group": "1",
          "id": "2",
          "start_time": 1540620000000,
          "title": "title 2",
        },
      ],
    },
    "2": Object {
      "group": Object {
        "id": "2",
      },
      "height": 30,
      "index": 1,
      "itemDimensions": Array [],
      "items": Array [],
    },
    "3": Object {
      "group": Object {
        "id": "3",
      },
      "height": 30,
      "index": 2,
      "itemDimensions": Array [
        Object {
          "dimensions": Object {
            "collisionLeft": 1540656000000,
            "collisionWidth": 20397548,
            "height": 22.5,
            "left": 8375,
            "stack": true,
            "top": 3.75,
            "width": 625,
          },
          "id": "3",
        },
      ],
      "items": Array [
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540676397548,
          "group": "3",
          "id": "3",
          "start_time": 1540656000000,
          "title": "title 3",
        },
      ],
    },
  },
  "height": 120,
  "itemsWithInteractions": Array [
    Object {
      "canMove": false,
      "canResize": false,
      "end_time": 1540550403877,
      "group": "1",
      "id": "0",
      "start_time": 1540540000000,
      "title": "title 0",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540554003877,
      "group": "1",
      "id": "5",
      "start_time": 1540532800000,
      "title": "title 5",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540575603877,
      "group": "1",
      "id": "6",
      "start_time": 1540550800000,
      "title": "title 6",
    },
    Object {
      "canMove": true,
      "canResize": "both",
      "end_time": 1540584875919,
      "group": "1",
      "id": "1",
      "start_time": 1540570000000,
      "title": "title 1",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540640397548,
      "group": "1",
      "id": "2",
      "start_time": 1540620000000,
      "title": "title 2",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540676397548,
      "group": "3",
      "id": "3",
      "start_time": 1540656000000,
      "title": "title 3",
    },
  ],
}
`;

exports[`stackItems work as expected 1`] = `
Object {
  "groupHeights": Array [
    60,
    30,
    30,
  ],
  "groupTops": Array [
    0,
    60,
    90,
  ],
  "groupsWithItemsDimensions": Object {
    "1": Object {
      "group": Object {
        "id": "1",
      },
      "height": 60,
      "index": 0,
      "itemDimensions": Array [
        Object {
          "dimensions": Object {
            "collisionLeft": 1540540000000,
            "collisionWidth": 6803877,
            "height": 22.5,
            "left": 4347.222222222223,
            "stack": true,
            "top": 3.75,
            "width": 236.24572916666602,
          },
          "id": "0",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540532800000,
            "collisionWidth": 21203877,
            "height": 22.5,
            "left": 4097.222222222223,
            "stack": true,
            "top": 33.75,
            "width": 736.245729166666,
          },
          "id": "5",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540550800000,
            "collisionWidth": 24803877,
            "height": 22.5,
            "left": 4722.222222222223,
            "stack": true,
            "top": 3.75,
            "width": 861.245729166666,
          },
          "id": "6",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540570000000,
            "collisionWidth": 14875919,
            "height": 22.5,
            "left": 5388.888888888889,
            "stack": true,
            "top": 33.75,
            "width": 516.5249652777784,
          },
          "id": "1",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540620000000,
            "collisionWidth": 20397548,
            "height": 22.5,
            "left": 7125,
            "stack": true,
            "top": 3.75,
            "width": 708.2481944444444,
          },
          "id": "2",
        },
      ],
      "items": Array [
        Object {
          "canMove": false,
          "canResize": false,
          "end_time": 1540546803877,
          "group": "1",
          "id": "0",
          "start_time": 1540540000000,
          "title": "title 0",
        },
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540554003877,
          "group": "1",
          "id": "5",
          "start_time": 1540532800000,
          "title": "title 5",
        },
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540575603877,
          "group": "1",
          "id": "6",
          "start_time": 1540550800000,
          "title": "title 6",
        },
        Object {
          "canMove": true,
          "canResize": "both",
          "end_time": 1540584875919,
          "group": "1",
          "id": "1",
          "start_time": 1540570000000,
          "title": "title 1",
        },
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540640397548,
          "group": "1",
          "id": "2",
          "start_time": 1540620000000,
          "title": "title 2",
        },
      ],
    },
    "2": Object {
      "group": Object {
        "id": "2",
      },
      "height": 30,
      "index": 1,
      "itemDimensions": Array [],
      "items": Array [],
    },
    "3": Object {
      "group": Object {
        "id": "3",
      },
      "height": 30,
      "index": 2,
      "itemDimensions": Array [
        Object {
          "dimensions": Object {
            "collisionLeft": 1540656000000,
            "collisionWidth": 20397548,
            "height": 22.5,
            "left": 8375,
            "stack": true,
            "top": 3.75,
            "width": 625,
          },
          "id": "3",
        },
      ],
      "items": Array [
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540676397548,
          "group": "3",
          "id": "3",
          "start_time": 1540656000000,
          "title": "title 3",
        },
      ],
    },
  },
  "height": 120,
  "itemsWithInteractions": Array [
    Object {
      "canMove": false,
      "canResize": false,
      "end_time": 1540546803877,
      "group": "1",
      "id": "0",
      "start_time": 1540540000000,
      "title": "title 0",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540554003877,
      "group": "1",
      "id": "5",
      "start_time": 1540532800000,
      "title": "title 5",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540575603877,
      "group": "1",
      "id": "6",
      "start_time": 1540550800000,
      "title": "title 6",
    },
    Object {
      "canMove": true,
      "canResize": "both",
      "end_time": 1540584875919,
      "group": "1",
      "id": "1",
      "start_time": 1540570000000,
      "title": "title 1",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540640397548,
      "group": "1",
      "id": "2",
      "start_time": 1540620000000,
      "title": "title 2",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540676397548,
      "group": "3",
      "id": "3",
      "start_time": 1540656000000,
      "title": "title 3",
    },
  ],
}
`;

exports[`stackItems work as expected no stack 1`] = `
Object {
  "groupHeights": Array [
    30,
    30,
    30,
  ],
  "groupTops": Array [
    0,
    30,
    60,
  ],
  "groupsWithItemsDimensions": Object {
    "1": Object {
      "group": Object {
        "id": "1",
      },
      "height": 30,
      "index": 0,
      "itemDimensions": Array [
        Object {
          "dimensions": Object {
            "collisionLeft": 1540540000000,
            "collisionWidth": 6803877,
            "height": 22.5,
            "left": 4347.222222222223,
            "stack": true,
            "top": 3.75,
            "width": 236.24572916666602,
          },
          "id": "0",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540532800000,
            "collisionWidth": 21203877,
            "height": 22.5,
            "left": 4097.222222222223,
            "stack": true,
            "top": 3.75,
            "width": 736.245729166666,
          },
          "id": "5",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540550800000,
            "collisionWidth": 24803877,
            "height": 22.5,
            "left": 4722.222222222223,
            "stack": true,
            "top": 3.75,
            "width": 861.245729166666,
          },
          "id": "6",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540570000000,
            "collisionWidth": 14875919,
            "height": 22.5,
            "left": 5388.888888888889,
            "stack": true,
            "top": 3.75,
            "width": 516.5249652777784,
          },
          "id": "1",
        },
        Object {
          "dimensions": Object {
            "collisionLeft": 1540620000000,
            "collisionWidth": 20397548,
            "height": 22.5,
            "left": 7125,
            "stack": true,
            "top": 3.75,
            "width": 708.2481944444444,
          },
          "id": "2",
        },
      ],
      "items": Array [
        Object {
          "canMove": false,
          "canResize": false,
          "end_time": 1540546803877,
          "group": "1",
          "id": "0",
          "start_time": 1540540000000,
          "title": "title 0",
        },
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540554003877,
          "group": "1",
          "id": "5",
          "start_time": 1540532800000,
          "title": "title 5",
        },
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540575603877,
          "group": "1",
          "id": "6",
          "start_time": 1540550800000,
          "title": "title 6",
        },
        Object {
          "canMove": true,
          "canResize": "both",
          "end_time": 1540584875919,
          "group": "1",
          "id": "1",
          "start_time": 1540570000000,
          "title": "title 1",
        },
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540640397548,
          "group": "1",
          "id": "2",
          "start_time": 1540620000000,
          "title": "title 2",
        },
      ],
    },
    "2": Object {
      "group": Object {
        "id": "2",
      },
      "height": 30,
      "index": 1,
      "itemDimensions": Array [],
      "items": Array [],
    },
    "3": Object {
      "group": Object {
        "id": "3",
      },
      "height": 30,
      "index": 2,
      "itemDimensions": Array [
        Object {
          "dimensions": Object {
            "collisionLeft": 1540656000000,
            "collisionWidth": 20397548,
            "height": 22.5,
            "left": 8375,
            "stack": true,
            "top": 3.75,
            "width": 625,
          },
          "id": "3",
        },
      ],
      "items": Array [
        Object {
          "canMove": false,
          "canResize": false,
          "className": "",
          "end_time": 1540676397548,
          "group": "3",
          "id": "3",
          "start_time": 1540656000000,
          "title": "title 3",
        },
      ],
    },
  },
  "height": 90,
  "itemsWithInteractions": Array [
    Object {
      "canMove": false,
      "canResize": false,
      "end_time": 1540546803877,
      "group": "1",
      "id": "0",
      "start_time": 1540540000000,
      "title": "title 0",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540554003877,
      "group": "1",
      "id": "5",
      "start_time": 1540532800000,
      "title": "title 5",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540575603877,
      "group": "1",
      "id": "6",
      "start_time": 1540550800000,
      "title": "title 6",
    },
    Object {
      "canMove": true,
      "canResize": "both",
      "end_time": 1540584875919,
      "group": "1",
      "id": "1",
      "start_time": 1540570000000,
      "title": "title 1",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540640397548,
      "group": "1",
      "id": "2",
      "start_time": 1540620000000,
      "title": "title 2",
    },
    Object {
      "canMove": false,
      "canResize": false,
      "className": "",
      "end_time": 1540676397548,
      "group": "3",
      "id": "3",
      "start_time": 1540656000000,
      "title": "title 3",
    },
  ],
}
`;
